fabio/openimage.py

```diff
@@ -219,27 +225,33 @@ def _openimage(filename):
                isinstance(file_obj.format, list):
                 # one of OXD/ADSC - should have got in previous
                 raise Exception("openimage failed on magic bytes & name guess")
-            filetype = file_obj.format
-
+            if file_obj.format is not None:
+                filetypes = [file_obj.format]
         except Exception:
             logger.debug("Backtrace", exc_info=True)
             raise IOError("Fabio could not identify " + filename)
 
-    if filetype is None:
+    if filetypes is None:
         raise IOError("Fabio could not identify " + filename)
 
-    klass_name = "".join(filetype) + 'image'
+    objs = []
+    for filetype in filetypes:
+        klass_name = "".join(filetype) + 'image'
+        try:
+            obj = fabioformats.factory(klass_name)
+            objs.append(obj)
+        except (RuntimeError, Exception):
+            pass
```

Comment:
We should a least log the fact there was an exception ...
Beside this  Runtime derives from Exception ...
```
In [4]: RuntimeError.__mro__
Out[4]: (RuntimeError, StandardError, Exception, BaseException, object)
```
So `except Exception as err` should be enough